src/app/auth/components/signup/signup.component.ts

```diff
@@ -110,14 +114,55 @@ export class SignupComponent {
         [Validators.required, Validators.minLength(MIN_PASSWORD_LENGTH)],
       ],
     });
-
     const confirmPasswordControl = new UntypedFormControl('', [
       Validators.required,
       matchControlValidator(this.signupForm.controls['password']),
     ]);
     this.signupForm.addControl('confirm', confirmPasswordControl);
   }
 
+  ngOnInit(): void {
+    this.signupForm.controls['password'].valueChanges.subscribe((password) => {
```

Comment:
Does this subscription get automatically cleaned up? It might, and I'd love confirmation that it does, but I feel like subscriptions have occasionally caused errors in the application once the things they refer to have been destroyed.